src/ui/storyTellingServerUI.js

```diff
@@ -0,0 +1,143 @@
+        results = regex.exec(url);
+    if (!results) { return null; }
+    if (!results[2]) { return ""; }
+    return decodeURIComponent(results[2].replace(/\+/g, " "));
+};
+
+/* Loads a story View page and a particular story from a story ID from the query string
+ * - "theme": the theme of the story View page (pass "page" in for base theme)
+ * - "options": additional options to merge into the View page
+ */
+sjrk.storyTelling.loadStoryFromParameter = function (theme, options) {
+    var storyPromise = fluid.promise();
+
+    var storyId = sjrk.storyTelling.getParameterByName("id");
+
+    if (storyId) {
+        $.get("/stories/" + storyId, function (data) {
+            var retrievedStory = JSON.parse(data);
+
+            options = options || {};
+            options.distributeOptions = {
+                "target": "{that story}.options.model",
+                "record": retrievedStory
+            };
+
+            var storyViewComponent = sjrk.storyTelling[theme].storyView(options);
+            storyPromise.resolve(storyViewComponent);
+        }).fail(function (error) {
+            storyPromise.reject(error);
+        });;
+    } else {
+        storyPromise.reject();
+    }
+
+    return storyPromise;
+};
+
+/* Loads a story Browse page and populates it with a set of stories
+ * - "theme": the theme of the story Browse page (pass "page" in for base theme)
+ * - "options": additional options to merge into the Browse page
+ */
+sjrk.storyTelling.loadBrowse = function (theme, options) {
+    var storiesPromise = fluid.promise();
+
+    $.get("/stories", function (data) {
+        var browseResponse = JSON.parse(data);
+
+        options = options || {};
+        options.distributeOptions = {
+            "target": "{that storyBrowser}.options.model",
+            "record": browseResponse
+        };
+
+        var storyBrowseComponent = sjrk.storyTelling[theme].storyBrowse(options);
+        storiesPromise.resolve(storyBrowseComponent);
+    }).fail(function (error) {
+        storiesPromise.reject(error);
+    });
+
+    return storiesPromise;
+};
+
+/* Loads custom theme files if and only if one of either the theme override or
+ * server configuration theme are present AND not set to the base theme.
+ * Once complete, the provided callback function is called.
+ * - "callback": a function to call once everything has completed. Will be called
+ *               regardless of whether theme information was specified or retrieved
+ * - "themeOverride": allows overriding of the theme stored in the configuration
+ */
+sjrk.storyTelling.loadThemedPage = function (callback, themeOverride) {
+    var loadPromise = fluid.promise();
+
+    var callbackFunction = typeof callback === "function" ? callback : fluid.getGlobalValue(callback);
+
+    $.get("/clientConfig").then(function (data) {
+        var theme = themeOverride ? themeOverride : data.clientConfig ? data.clientConfig.theme : "base";
```

Comment:
The fall-back logic to the base theme should be on the server side as the first thing when global config is read in, which can then be used by the following code as a proper theme. It's better to handle this kind of case as early as possible.